### PR TITLE
fix: git server url is hardcoded in pipelinerunner #4588

### LINKED ISSUE #4588: pipelinerunner doesnot support github domain other than "github.com"
### Summary 
My enterprise github is like "github.xxx.xxx.com".
Current pipelinerunner only supports "github.com" which is hard-coded.
[controller_pipelinerunner](https://github.com/jenkins-x/jx/blob/master/pkg/cmd/controller/pipelinerunner.go#L282)

### Expected behavior

By introducing new cmd option `--git-provider-url=xx.xxx.com`,  github domain can be set.
And the clone operation is successful.

### Actual behavior
When the app is created by `jx create  --git-provider-url=xx.xxx.com`, 
pipelinerrunner pod will exit and be restarted due to checkout error for more than 3 times.

### Jenkins type

- [+ ] Serverless Jenkins X Pipelines (Tekton + Prow)
- [ ] Classic Jenkins

### PR DESCRIPTION
**Pull request by rokii**
#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Which issue this PR fixes

fixes #4588

### CODE REVIEW

File: pkg/cmd/controller/pipelinerunner.go [hunk 1]
```diff
@@ -279,7 +283,10 @@ func (o *PipelineRunnerOptions) startPipeline(pipelineRun PipelineRunRequest) (P
 		return response, errors.Wrap(err, "failed to get env vars from prowjob")
 	}
 
-	sourceURL := fmt.Sprintf("https://github.com/%s/%s.git", prowJobSpec.Refs.Org, prowJobSpec.Refs.Repo)
+	sourceURL := o.getSourceURL(prowJobSpec.Refs.Org, prowJobSpec.Refs.Repo)
+	if sourceURL == "" {
```
Comment (hferentschik) [inline pkg/cmd/controller/pipelinerunner.go]: +1 for falling back to github

File: pkg/cmd/controller/pipelinerunner.go [hunk 2]
```diff
@@ -457,3 +464,42 @@ func (o *PipelineRunnerOptions) setupSignalChannel(cancel context.CancelFunc) {
 		cancel()
 	}()
 }
+
+func (o *PipelineRunnerOptions) getSourceURL(org, repo string) string {
+	jxClient, ns, err := o.getClientsAndNamespace()
```
Comment (hferentschik) [inline pkg/cmd/controller/pipelinerunner.go]: You are missing an err check here. _jxClient_ might be nil here.

File: pkg/cmd/controller/pipelinerunner.go [hunk 3]
```diff
@@ -457,3 +464,42 @@ func (o *PipelineRunnerOptions) setupSignalChannel(cancel context.CancelFunc) {
...
+
+	resourceInterface := jxClient.JenkinsV1().SourceRepositories(ns)
```
Comment (hferentschik) [inline pkg/cmd/controller/pipelinerunner.go]: given that all you need _jxClient_ in this function is for getting the source repository interface, you might as well push that into `getClientsAndNamespace` and call it `getSourceRepositoryInterface`. Up to you really.

**Comment by rokii** [comment]
I can not access the integration test dashboard. 
```
Failed to authenticate: github: user "rokii" not in required orgs or teams
```

File: pkg/cmd/controller/pipelinerunner.go (hunk 2 above)
Comment (rokii) [reply pkg/cmd/controller/pipelinerunner.go]: done

File: pkg/cmd/controller/pipelinerunner.go (hunk 3 above)
Comment (rokii) [reply pkg/cmd/controller/pipelinerunner.go]: `getClientsAndNamespace` is much general that may be used for some other case later.  So I wanna keep it.

**Comment by hferentschik** [comment]
/lgtm

Thanks @rokii

**Pull request merged**